src/variableParser.ts

```diff
@@ -38,23 +38,47 @@ export function getEnvironments(bundleContent: string | undefined, varFileConten
   return [...uniqueNames];
 }
 
+function replaceEnvironmentVariables(vars: Variables): {
+  replacedVars: Variables;
+  undefinedVars: string[];
+} {
+  const PREFIX = "$env.";
+
+  const undefinedVars: string[] = [];
+  const getVal = (val: any): any => {
+    if (typeof val !== "string" || !val.startsWith(PREFIX)) return val;
+
+    const envVarName = val.slice(PREFIX.length);
+    if (envVarName in process.env) return process.env[envVarName];
```

Comment:
Perhaps a bad idea.
```
myVar: $env.MY_VAR ?? TEST
```

Reply:
I'm not sure what you mean. An assumption of the feature is that `$env.X` instances will only be in the value of variable declarations, and that they will be the entire value. 

In the above example you've shown, the variable name we search for would be `"MY_VAR ?? TEST"`, for example. Since we can set essentially any attribute name in `process.env`, I feel this offers the most flexibility.

Reply:
Sorry, wasn't clear. Think JS/TS: if $env_MY_VAR is null or undefined, default to TEST.

I'm suggesting we could allow defining a default if the environment variable is undefined.

Reply:
Sounds like a plan.